apps/vercel/src/clients/Vercel.ts

```diff
@@ -0,0 +1,76 @@
+import { CreateDeploymentInput, Deployment, ListProjectsResponse } from '../types';
+
+interface VercelAPIClient {
+  checkToken: () => Promise<boolean>;
+  listProjects: () => Promise<ListProjectsResponse>;
+  createDeployment: (input: CreateDeploymentInput) => Promise<Deployment>;
+  getDeploymentById: (deploymentId: string) => Promise<Deployment>;
+}
+
+export default class VercelClient implements VercelAPIClient {
```

Comment:
I guess we should discuss moving this to the backend (but not for now, not for this PR)